### PR TITLE
Kafka cluster CRD spec was updated with Kraft field

### PR DESCRIPTION
**Pull request by tengu-alt**

### CODE REVIEW

File: pkg/models/errors.go [hunk 1]
```diff
@@ -33,6 +35,7 @@ var (
 	ErrTypeAssertion                         = errors.New("unable to assert type")
 	ErrImmutableSchemaRegistry               = errors.New("schema registry is immutable")
 	ErrImmutableRestProxy                    = errors.New("rest proxy is immutable")
+	ErrImmutableKraft                        = errors.New("Kraft is immutable")
```
Comment (testisnullus) [inline pkg/models/errors.go:38]: Kraft -> kraft

File: pkg/models/errors.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import "errors"
 
 var (
 	ErrZeroDataCentres                       = errors.New("cluster spec doesn't have data centres")
+	ErrMoreThanOneKraft                      = errors.New("cluster spec spec does not support more than one Kraft")
+	ErrMoreThanThreeControllerNodeCount      = errors.New("Kraft does not support more than three controller nodes")
```
Comment (testisnullus) [inline pkg/models/errors.go:24]: Kraft -> kraft

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: pkg/models/kafka_apiv2.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ type KafkaCluster struct {
 	DedicatedZookeeper                []*DedicatedZookeeper     `json:"dedicatedZookeeper"`
 	DefaultNumberOfPartitions         int                       `json:"defaultNumberOfPartitions"`
 	DefaultReplicationFactor          int                       `json:"defaultReplicationFactor"`
+	Kraft                             []*Kraft                  `json:"kraft"`
```
Comment (worryg0d) [inline pkg/models/kafka_apiv2.go:37]:
```suggestion
	Kraft                             []*Kraft                  `json:"kraft,omitempty"`
```

File: pkg/models/errors.go (hunk 1 above)
Comment (tengu-alt) [reply pkg/models/errors.go]: done

File: pkg/models/errors.go (hunk 2 above)
Comment (tengu-alt) [reply pkg/models/errors.go]: refactored

File: pkg/models/errors.go [hunk 4]
```diff
@@ -20,6 +20,8 @@ import "errors"
 
 var (
 	ErrZeroDataCentres                       = errors.New("cluster spec doesn't have data centres")
+	ErrMoreThanOneKraft                      = errors.New("cluster spec spec does not support more than one kraft")
```
Comment (ribaraka) [inline pkg/models/errors.go:23]: spec spec

**Review by ribaraka — APPROVED** [review]

File: pkg/models/errors.go (hunk 4 above)
Comment (tengu-alt) [reply pkg/models/errors.go]: refactored

**Review by testisnullus — APPROVED** [review]

**Review by worryg0d — APPROVED** [review]

**Pull request merged**